File: hashing-subset/isSubset.java
 class HashingSubset {

     static boolean isSubset(int[] arrayOne, int[] arrayTwo) {
        // Iterate over each element in the second array
        int arrayOneLength = arrayOne.length;
        int arrayTwoLength = arrayTwo.length;
        for (int i = 0; i < arrayTwoLength; i++) {
            boolean found = false;
            // Check if the element exists in the first array
            for (int j = 0; j < arrayOneLength; j++) {
                if (arrayTwo[i] == arrayOne[j]) {
                    found = true;
                    break;
                }
            }
            // If any element is not found, return false
            if (!found) {
                return false;
            }
        }
        // If all elements are found, return true
        return true;
    }

    public static void main(String[] args) {
        int[] a = {11, 1, 13, 21, 3, 7};
        int[] b = {11, 3, 7, 1};

        if (isSubset(a, b)) {
            System.out.println("true");
        } else {
            System.out.println("false");
        }
    }
}

File: hashing-subset/isSubset.js
function isSubset(arrayOne, arrayTwo) {

    const arrayOneLength = arrayOne.length;
    const arrayTwoLength = arrayTwo.length;
    // Iterate over each element in the second array
    for (let arrTwoIndex = 0; arrTwoIndex < arrayTwoLength; arrTwoIndex++) {
      let found = false;
      
      // Check if the element exists in the first array
      for (let arrayTwoIndex = 0; arrayTwoIndex < arrayOneLength; arrayTwoIndex++) {
        if (arrayTwo[arrTwoIndex] === arrayOne[arrayTwoIndex]) {
          found = true;
          break;
        }
      }
      
      // If any element is not found, return false
      if (!found) {
        return false;
      }
    }
    
    // If all elements are found, return true
    return true;
  }
  
  //Driver Code
  let arrayOne = [11, 1, 13, 21, 3, 7];
  let arrayTwo = [11, 3, 7, 1];
  
  if (isSubset(arrayOne, arrayTwo)) {
    console.log("true");
  } else {
    console.log("false");
  }
  